### PR TITLE
Add a RUNPATH to installed libraries

### PR DESCRIPTION
**Pull request by PatKamin**
### Description

This is a cherry-picked patch merged to the v0.9 branch: #882 

### Checklist

- [x] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -115,6 +115,9 @@ elseif(MACOSX)
 endif()
 
 if(UMF_BUILD_SHARED_LIBRARY)
+    # Set the runtime search path to the directory containing hwloc library
+    set(CMAKE_INSTALL_RPATH "\$ORIGIN")
```
Comment (lukaszstolarczuk) [inline src/CMakeLists.txt:119]: just want to make sure - isn't there a better solution to that problem...? 😉 now, some random value of `$ORIGIN` could make unexpected issues, or something...?

**Review by lukaszstolarczuk — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (lukaszstolarczuk) [reply src/CMakeLists.txt:119]: we can do similarly like we have for hwloc name change and only set it when required

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -115,6 +115,10 @@ elseif(MACOSX)
 endif()
 
 if(UMF_BUILD_SHARED_LIBRARY)
+    if(NOT "${UMF_INSTALL_RPATH}" STREQUAL "")
+        set(CMAKE_INSTALL_RPATH ${UMF_INSTALL_RPATH})
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply src/CMakeLists.txt]: Added

File: CMakeLists.txt [hunk 3]
```diff
@@ -82,6 +82,12 @@ set(UMF_PROXY_LIB_BASED_ON_POOL
 set_property(CACHE UMF_PROXY_LIB_BASED_ON_POOL
              PROPERTY STRINGS ${KNOWN_PROXY_LIB_POOLS})
 
+set(UMF_INSTALL_RPATH
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply CMakeLists.txt]: Moved

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -115,6 +115,10 @@ elseif(MACOSX)
 endif()
 
 if(UMF_BUILD_SHARED_LIBRARY)
+    if(NOT "${UMF_INSTALL_RPATH}" STREQUAL "")
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply src/CMakeLists.txt]: Yes, it suffices, thanks.

File: CMakeLists.txt [hunk 5]
```diff
@@ -82,6 +82,12 @@ set(UMF_PROXY_LIB_BASED_ON_POOL
...
+    ""
+    CACHE
+        STRING
+        "Set the runtime search path to the directory containing hwloc library")
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply CMakeLists.txt]: Now it's clearer, thanks.

**Review by lukaszstolarczuk — APPROVED** [review]

File: CMakeLists.txt [hunk 6]
```diff
@@ -63,6 +63,11 @@ option(UMF_FORMAT_CODE_STYLE
 set(UMF_HWLOC_NAME
     "hwloc"
     CACHE STRING "Custom name for hwloc library w/o extension")
+    set(UMF_INSTALL_RPATH
```
[the remark this answers is not in the document]
Comment (PatKamin) [reply CMakeLists.txt]: Fixed

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**